src/sound_openal.cpp

```diff
@@ -290,68 +349,28 @@ class OpenALSoundManager: public ISoundManager
 	std::unordered_map<int, FadeState> m_sounds_fading;
 	float m_fade_delay;
 public:
-	bool m_is_initialized;
-	OpenALSoundManager(OnDemandSoundFetcher *fetcher):
+	OpenALSoundManager(SoundManagerGlobal *smg, OnDemandSoundFetcher *fetcher):
 		m_fetcher(fetcher),
-		m_device(NULL),
-		m_context(NULL),
+		m_device(smg->m_device.get()),
+		m_context(smg->m_context.get()),
 		m_next_id(1),
-		m_fade_delay(0),
-		m_is_initialized(false)
+		m_fade_delay(0)
 	{
-		ALCenum error = ALC_NO_ERROR;
-
 		infostream<<"Audio: Initializing..."<<std::endl;
-
-		m_device = alcOpenDevice(NULL);
-		if(!m_device){
-			infostream<<"Audio: No audio device available, audio system "
-				<<"not initialized"<<std::endl;
-			return;
-		}
-
-		m_context = alcCreateContext(m_device, NULL);
-		if(!m_context){
-			error = alcGetError(m_device);
-			infostream<<"Audio: Unable to initialize audio context, "
-					<<"aborting audio initialization ("<<alcErrorString(error)
-					<<")"<<std::endl;
-			alcCloseDevice(m_device);
-			m_device = NULL;
-			return;
-		}
-
-		if(!alcMakeContextCurrent(m_context) ||
-				(error = alcGetError(m_device) != ALC_NO_ERROR))
-		{
-			infostream<<"Audio: Error setting audio context, aborting audio "
-					<<"initialization ("<<alcErrorString(error)<<")"<<std::endl;
-			alcDestroyContext(m_context);
-			m_context = NULL;
-			alcCloseDevice(m_device);
-			m_device = NULL;
-			return;
-		}
-
-		alDistanceModel(AL_INVERSE_DISTANCE_CLAMPED);
-
-		infostream<<"Audio: Initialized: OpenAL "<<alGetString(AL_VERSION)
-				<<", using "<<alcGetString(m_device, ALC_DEVICE_SPECIFIER)
-				<<std::endl;
-
-		m_is_initialized = true;
+		infostream<<"Audio: Initialized: OpenAL "<<std::endl;
 	}
 
 	~OpenALSoundManager()
 	{
 		infostream<<"Audio: Deinitializing..."<<std::endl;
-		// KABOOM!
-		// TODO: Clear SoundBuffers
-		alcMakeContextCurrent(NULL);
-		alcDestroyContext(m_context);
-		m_context = NULL;
-		alcCloseDevice(m_device);
-		m_device = NULL;
+
+		std::set<int> source_del_list;
```

Comment:
order is not important, please use std::unordered_set

Reply:
I've followed convention for source_del_list that was preexisting in the source file.
The other use of source_del_list will be converted to unordered as well.